core/crawler.py

```
@@ -1,70 +1,67 @@
+    """
+
+    links = [] 
+    pagesToVisit = [url]
+    numberVisited = 0
+    while numberVisited < maxPages and pagesToVisit:
+        numberVisited += 1
+        url = pagesToVisit.pop()
+        try:
+            parser = LinkParser()
+            data, links = parser.getLinks(url)
+        except:
```

Ruff (0.15.9)

[error] Do not use bare `except`

(E722)